tasmota/xsns_63_aht1x.ino

```diff
@@ -26,142 +26,198 @@
+  return (!isnan(aht1x_sensors[aht1x_idx].temperature) && !isnan(aht1x_sensors[aht1x_idx].humidity) && (aht1x_sensors[aht1x_idx].humidity != 0));
 }
 
 /********************************************************************************************/
 
-bool AHT10Init(void)
+// Test for Polling the device without delays
+// Incompatible with other devices on I2C bus
+void AHT1XPoll(void) // We have 100ms for read. Sensor needs 80-95 ms
 {
-  Wire.begin(AHT10_ADDR);
-  Wire.beginTransmission(AHT10_ADDR);
-  Wire.write(eSensorCalibrateCmd, 3); // init with internal temp coef.
-  Wire.endTransmission();
-
-  delay(40);  //  after tests, its ok
-
-  return (0x08 == (AHT10ReadStatus() & 0x68)); 
+  aht1x_Pcount++;
+   switch (aht1x_Pcount) {
+    case 10:
+      AHT1XWrite(0);
+      break;
+    case 11:
+      if (AHT1XRead(0)){
+        ConvertTemp(aht1x_sensors[0].temperature);   // Set global temperature
+        ConvertHumidity(aht1x_sensors[0].humidity);  // Set global humidity
+      }
+      aht1x_Pcount = 0;
+      break;
+    }
 }
 
-uint8_t AHT10ReadStatus(void)
+unsigned char AHT1XReadStatus(uint8_t aht1x_address)
 {
-  Wire.requestFrom(AHT10_ADDR, 1);
-  uint8_t result = Wire.read();
+  uint8_t result = 0;
+  Wire.requestFrom(aht1x_address, (uint8_t) 1);
+  result = Wire.read();
   return result;
 }
 
-void AHT10Reset(void)
+void AHT1XReset(uint8_t aht1x_address)
 {
-  Wire.beginTransmission(AHT10_ADDR);
-  Wire.write(eSensorResetCmd);
+  Wire.beginTransmission(aht1x_address);
+  Wire.write(AHTResetCmd);
   Wire.endTransmission();
-  delay(20);
+  delay(AHT1X_RST_DELAY);
 }
 
 /********************************************************************************************/
-
-void AHT10Detect(void)
+bool AHT1XInit(uint8_t aht1x_address)
 {
-  if (I2cActive(AHT10_ADDR)) { return; }
+  Wire.beginTransmission(aht1x_address);
+  Wire.write(AHTSetCalCmd, 3);
+  if (Wire.endTransmission() != 0) return false;
+  delay(AHT1X_CMD_DELAY);
+  if((AHT1XReadStatus(aht1x_address) & 0x68) == 0x08) // Sensor calcoef aktiv
+    return true;
+  return false;
+}
 
-  if (AHT10Init()) {
-    I2cSetActiveFound(AHT10_ADDR, AHT10.name);
-    AHT10.count = 1;
+void AHT1XDetect(void)
+{
+  for (uint8_t i = 0; i < AHT1X_MAX_SENSORS; i++) {
+    if (I2cActive(aht1x_addresses[i])) { continue; }
+    if (AHT1XInit(aht1x_addresses[i]))
+    {
+      aht1x_sensors[aht1x_count].address = aht1x_addresses[i];
+      GetTextIndexed(aht1x_sensors[aht1x_count].types, sizeof(aht1x_sensors[aht1x_count].types), i, ahtTypes);
+      I2cSetActiveFound(aht1x_sensors[aht1x_count].address, aht1x_sensors[aht1x_count].types);
+      aht1x_count = 1;
+      break; // Only one Sensor allowed at the moment (I2C Sensor-Bug)
+    }
   }
 }
 
-void AHT10EverySecond(void)
+float CalcTemHumToDew(float h, float t)
 {
-  if (uptime &1) {
-    // AHT10: 55mS
-    if (!AHT10Read()) {
-      AddLogMissed(AHT10.name, AHT10.valid);
-    }
+  float gamma = log(h / 100) + 17.62 * t / (243.5 + t);
```

Comment:
Could you try the below code if it reports the same result:
```
  float gamma = TaylorLog(h / 100) + 17.62 * t / (243.5 + t);
```
This would save 1k5 code space as Tasmota hates ``log()`` from the math library.

Reply:
Code is working with Taylorlog!

Reply:
Great! Already released for all temp/hum sensors.